android/src/main/java/com/rnfs/RNFSManager.java

```diff
@@ -759,6 +765,80 @@ public void touch(String filepath, double mtime, double ctime, Promise promise)
     }
   }
 
+  @ReactMethod
+  public void canOpenFile(String path, String mime, final Promise promise) {
+    try {
+      Uri uriForFile = FileProvider.getUriForFile(getCurrentActivity(),
+              this.getReactApplicationContext().getPackageName() + ".provider", new File(path));
+
+      // Create the intent with data and type
+      Intent intent = new Intent(Intent.ACTION_VIEW)
+              .setDataAndType(uriForFile, mime);
+
+      // Set flag to give temporary permission to external app to use FileProvider
+      intent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+        // All the activity to be opened outside of an activity
+      intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+      // Validate that the device can open the file
+      PackageManager pm = getCurrentActivity().getPackageManager();
+      if (intent.resolveActivity(pm) != null) {
+        promise.resolve(true);
+      }
+      else {
+        promise.resolve(false);
+      }
+
+    } catch(Exception ex) {
+      promise.reject("EUNSPECIFIED", ex.getLocalizedMessage());
+    }
+  }
+
+  @ReactMethod
+  public void openFile(String path, String mime, final Promise promise) {
+    try {
+      Uri uriForFile = FileProvider.getUriForFile(getCurrentActivity(),
+              this.getReactApplicationContext().getPackageName() + ".provider", new File(path));
+
+      // Create the intent with data and type
+      Intent intent = new Intent(Intent.ACTION_VIEW)
+              .setDataAndType(uriForFile, mime);
+
+      // Set flag to give temporary permission to external app to use FileProvider
+      intent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+        // All the activity to be opened outside of an activity
+      intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+      // Validate that the device can open the file
+      PackageManager pm = getCurrentActivity().getPackageManager();
+      if (intent.resolveActivity(pm) != null) {
+        this.getReactApplicationContext().startActivity(intent);
+      }
+
+      ActionViewVisible = true;
```

Comment:
Some of the logic can be de-duped. Kept it the same atm to compare to the original.